compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion/Primitives.hs

```diff
@@ -373,6 +373,25 @@ convertPrim version "EToAnyContractKey"
         ETmLam (mkVar "key", key) $
         EToAny key (EVar $ mkVar "key")
 
+-- Exceptions
+convertPrim _ "BEAnyExceptionMessage" (TBuiltin BTAnyException :-> TText) =
+    EBuiltin BEAnyExceptionMessage
+convertPrim _ "BEGeneralErrorMessage" (TBuiltin BTGeneralError :-> TText) =
+    EBuiltin BEGeneralErrorMessage
+convertPrim _ "BEArithmeticErrorMessage" (TBuiltin BTArithmeticError :-> TText) =
+    EBuiltin BEArithmeticErrorMessage
+convertPrim _ "BEContractErrorMessage" (TBuiltin BTContractError :-> TText) =
+    EBuiltin BEContractErrorMessage
+
+-- TODO #8020 https://github.com/digital-asset/daml/issues/8020
+-- Handle these three in LFConversion.hs and check that ty1 is an exception type.
```

Comment:
What exactly do we need in `LFConversion.hs` for these?

Reply:
We need some easy way to refer to them in LFConversion (giving them separate names like "magicThrow" and "magicToAnyException" etc). It's just a little more involved. But the idea is we could then prevent "magicThrow", etc from being invoked at non-LF exception types.